src/Runtime/OMTensorList.inc

```diff
@@ -103,23 +116,18 @@ void omTensorListDestroy(OMTensorList *list) {
     return;
   for (int64_t i = 0; i < list->_size; i++)
     omTensorDestroy(list->_omts[i]);
-  if (list->_owning) {
-    free(list->_omts);
-  }
+  // Free the list as well as the pointers to the OMTensor array
```

Comment:
Could this just call `omTensorListDestroyShallow(list)` instead of replicating the two free calls? That should keep things easier if we ever expand TensorList and need additional cleanup in the future.

Reply:
Okay I updated it